src/rofi_rbw/typer.py

```diff
@@ -75,3 +75,34 @@ def type_characters(self, characters: str, active_window: str) -> None:
             'wtype',
             characters
         ])
+
+class YDotoolTyper(Typer):
+    @staticmethod
+    def supported() -> bool:
+        return is_wayland() and is_installed('ydotool')
+
+    @staticmethod
+    def name() -> str:
+        return 'ydotool'
+
+    def get_active_window(self) -> str:
+        return "not possible with ydotool"
+
+    def type_characters(self, characters: str, active_window: str, use_root = False) -> None:
+        if use_root:
+            run([
+                'sudo',
+                'ydotool',
+                'type',
+                characters
+            ])
+        else:
+            # The delay is needed for it to work, because the compositor may need some time to register a new input device. Without delay, the first few keys have a chance to be missed.
+            run([
+                'ydotool',
+                'sleep',
+                '50',
```

Comment:
The delay time probably should also be tweakable.

Reply:
In general, less options is usually nicer. As the delay is only a technical requirement, I don't expect that people want to change that, as long as it works.